### ISSUE TITLE
Does opentsdb support post annotation with metric_name and tags?

### ISSUE THREAD

**Issue by hzhaop** [OpenTSDB/opentsdb#913]
hi, 
I am new to opentsdb, and I go through the apis of opentsdb but I cannot find a way to post annotation with metric_name and tags, instead of with tsuid. 
That could be a performance problem because every time I want to post an annotation, I have to look up the tsuid of the metric_name and tags and cache into local memory so that next time when the same metric and tags come, I could find the tsuid immediately(not through the http api).  This may consume a lot of work and memory.  
if not , is there any plan to realize or some workaround method？

**Comment by manolama** [comment]
I dont think we ever added this specifically. I'll mark it as a feature request and see if we can get it into 2.4. Shouldn't be difficult now. (The original idea was to only allow annotations for series that had data points but some folks don't care about numbers and just want the strings)

**Comment by hzhaop** [comment]
@manolama  Thanks, I'm very looking forward to this feature.

**Comment by hzhaop** [comment]
And  I want to make a suggestion if someone would like to implement this. In my opinion, this feature can be merged to the endpoint of /api/put so folks are able to add data points and annotation at the same time.

**Comment by lordang** [comment]
I need string as data value too. I considered to use annotation instead, but it doesn't support post annotation with metric_name and tags. only support tsuid.
If you guys have plan to support string as data value, we're very appreciated.

**Comment by bipulkdas** [comment]
Hi, I am also working on a use case that requires posting both metric value and strings as part of the same api call. Currently the way it can be done is to look up the tsuids and then post an annotation for the tsuids. would greatly appreciate if you support the feature in the next release.

**Comment by Harshvardhan28** [comment]
@manolama We can't do this in v2.4.0 or it is done?
